HW04.Rmd

```diff
@@ -1,126 +1,273 @@
+
+```{r month_chart}
+#change variable month to contain abbs of months
+gun_death$month <- factor(gun_death$month, labels= c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"))
+
+#create a data frame that has the count of death for each month 
+monthly_death<- gun_death %>%
+    group_by(month) %>%
+    count()
+
+#plot   
+ggplot(monthly_death, aes(x=month, y=n))+
+    geom_bar(stat="identity", fill="red", alpha=0.7)+
+    scale_y_continuous(limit=c(0,12000))+
+    labs(title="Montly Gun Deaths in 2012", x=element_blank(), y="Number of deaths")+
+    theme_bw()
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+#create a data frame with death number associated with intent type
+gun_death_intent<- gun_death %>%
+    group_by(intent) %>%
+    count() 
+
+#A way to level the intent variable in order to order the bins, not the best way if there are more variables
+gun_death_intent$intent<- factor(gun_death_intent$intent, 
+                                 levels=c("Suicide", "Homicide", "Accidental", "Undetermined", "Not Available"))     
+
+#plot
+ggplot(gun_death_intent, aes(x=intent, y=n, fill=intent))+
+    geom_bar(stat="identity", alpha=0.9)+
+    scale_y_continuous(limit=c(0,64000))+
+    labs(title="Number of Gun Deaths by Different Intentions in 2012", x=element_blank(), y="Number of deaths", fill="Reported Intentions")+
+    theme_bw()
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+##make age variable numeric
+gun_death$age<- as.numeric(gun_death$age)
+
+#mean of female victims
+female_age<- gun_death %>%
+    select(age, sex) %>%
+    filter(sex=="F") 
+
+avg<- mean(female_age$age, na.rm=TRUE)  
+avg<- round(avg, digits=2)
+
+#plot
+ggplot(gun_death, aes(x=sex, y=age, fill=sex))+
+    geom_boxplot()+
+    annotate("text", x=1, y=70, label=paste("The average age\n of female victim is", avg))+
+    labs(title="Ages of Gun Violence Victims in 2012, By Gender", x=element_blank(), y="Age", fill="Gender")+
+    theme_bw()
+```
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+-The answer is 46227 
+
+```{r education}
+answer<- gun_death %>%
+    select(education, race, sex) %>%
+    filter(education %in% c("HS/GED", "Some college", "BA+"), race== "White", sex=="M") %>%
+    count()
+answer
+```
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+-Summer has the most gun deaths.
+
+```{r season}
+#calculate death count in each season
+winter<- gun_death %>%
+    select(month) %>%
```

Comment:
I actually didn't know where and when should I apply select function in my code, but from here I noticed that it can be very useful if it is used to indicate a certain category. So clever.